R/cut.R

```diff
@@ -0,0 +1,105 @@
+#' Cut a numeric Crunch variable
+#' 
+#' crunch::cut() is equivalent to base::cut() except that it operates on 
+#' crunch variables instead of in-memory variables. The function divides the range of 
+#' x into intervals and codes the values in x according to which interval they fall. 
+#' The leftmost interval corresponds to level one, the next leftmost to level two and 
+#' so on.
+#' @param x A crunch variable of the class NumericVariable
+#' @param breaks Either a numeric vector of two or more unique cut points 
+#' or a single number (greater than or equal to 2) giving the number of intervals 
+#' into which x is to be cut.
+#' @param variable.name The name of the resulting case variable as a character string. 
+#' @param labels labels for the levels of the resulting category.
+#' By default, labels are constructed using interval notation.
+#' If labels = FALSE, simple integer codes are returned instead of a factor.
+#' @param include.lowest logical, indicating if an `x[i]` equal to the lowest 
+#' (or highest, for right = FALSE) `breaks` value should be included.
+#' @param right logical, indicating if the intervals should be closed on the right 
+#' (and open on the left) or vice versa.
+#' @param dig.lab	integer which is used when labels are not given. 
+#' It determines the number of digits used in formatting the break numbers.
+#' @param ordered_result	Ignored.
+#' @param ... further arguments passed to or from other methods.
+#'
+#'
+#' @return a Crunch VariableDefinition
+#' @export
+#'
+#' @examples
+#' \dontrun{
+#' ds <- loadDataset("mtcars")
+#' ds$cat_var <- cut(ds$mpg, 3, variableName = "new_var")
+#' }
+#'  
+setMethod("cut", "NumericVariable", function(x, 
+                                             breaks, 
+                                             variable.name, 
```

Comment:
Check out how `combine` works for the variable metadata attributes: https://github.com/Crunch-io/rcrunch/blob/master/R/combine-categories.R#L28

A default "name" is given if one isn't supplied in `...` based on the number of categories in the result--you can do the same here. Other attributes not passed in `...` are brought from the original variable via `copyVariableReferences`.